src/phongo_bson.c

```diff
@@ -1156,6 +1201,9 @@ static bool php_phongo_bson_state_parse_type(zval* options, const char* name, ph
 	if (!strcasecmp(classname, "array")) {
 		element->type = PHONGO_TYPEMAP_NATIVE_ARRAY;
 		element->ce   = NULL;
+	} else if (!strcasecmp(classname, "bson")) {
```

Comment:
IIRC, the string constants are only used here where we parse a user-supplied type map, but the PHP code uses them extensively. However, if we want to expose these strings as constants for the benefit of users, I'd much rather use an `enum` than global constants (since we don't really have a class to attach them to).

Reply:
I was only referring to internal constants to avoid string literals in C. I don't have any desire to create PHP constants, as we don't really have a class to stuff them on, and I'm not excited about introducing global constants.